### PR TITLE
Add donate page

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
@wesm I copied some text from the pandas2 design docs (and also linked to those docs)

The intro is stolen from http://jupyter.org/donate.html

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
A preview can be seen here: https://pandas-dev.github.io/pandas-website/pr-donate/donate.html

(there are now two options for the link (iframe and button), was just trying out, if somebody has a preference)

File: donate.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+* Fixing long-standing limitations or inconsistencies in missing data (null values in integer and boolean data, and a more consistent notion of null / NA).
+* Improved performance and utilization of multicore systems.
+* Better user control / visibility of memory usage.
+* Clearer semantics around non-NumPy data types, and permitting new pandas-only data types to be added.
+* Exposing a “libpandas” C/C++ API to other Python library developers.
+
+More information about this can be found in the "pandas 2.0" design documents: https://pandas-dev.github.io/pandas2/index.html
+
+Donations are managed by the `NUMFocus <http://www.numfocus.org/open-source-projects.html>`__ Foundation.
```
Comment (chris-b1) [inline donate.rst]: I would move this text just below the opening paragraph, and also add a note about 501(c)3 / US tax deductibility.  The Jupyter language is pretty good, could use that.

"Our donations are managed by the NumFOCUS Foundation, which is the legal and fiscal umbrella for the project. NumFOCUS is a 501(c)3 non-profit foundation, so if you are subject to US Tax law, your contributions will be tax-deductible."

http://jupyter.org/donate.html

Comment (jorisvandenbossche) [inline donate.rst]: Good idea, added the full part about numfocus

**Comment by jreback** [comment]
This is great @jorisvandenbossche

**Comment by wesm** [comment]
LGTM, thanks @jorisvandenbossche!

**Pull request merged**